resources/js/charts/changelog-chart.ts

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import * as d3 from 'd3';
+import { first, groupBy, kebabCase, last, map, sumBy } from 'lodash';
+import moment from 'moment';
+import { fadeIn, fadeOut } from 'utils/fade';
+import { formatNumber } from 'utils/html';
+import { parseJson } from 'utils/json';
+
+interface BuildHistory {
+  created_at: string;
+  label: string | null;
```

Comment:
`null` label isn't supposed to happen. If it does, the data is wrong in the first place.